src/DynamicRupture/Output/Base.cpp

```diff
@@ -214,77 +216,329 @@ void Base::updateElementwiseOutput() {
+        ruptureVelocity(level, i) = this->computeRuptureVelocity();
+      }
+
+      auto& peakSlipsRate = std::get<VariableID::PeakSlipRate>(outputData.vars);
+      if (peakSlipsRate.isActive) {
+        auto* peakSR = local.layer->var(drDescr->peakSlipRate);
+        peakSlipsRate(level, i) = peakSR[local.ltsId][local.nearestGpIndex];
+      }
+
+      auto& dynamicStressTime = std::get<VariableID::DynamicStressTime>(outputData.vars);
+      if (dynamicStressTime.isActive) {
+        auto* dynStressTime = (local.layer->var(drDescr->dynStressTime));
+        dynamicStressTime(level, i) = dynStressTime[local.ltsId][local.nearestGpIndex];
+      }
+
+      auto& slipVectors = std::get<VariableID::Slip>(outputData.vars);
+      if (slipVectors.isActive) {
+        VrtxCoords crossProduct = {0.0, 0.0, 0.0};
+        MeshTools::cross(strike, tangent1, crossProduct);
+
+        double cos1 = MeshTools::dot(strike, tangent1);
+        double scalarProd = MeshTools::dot(crossProduct, normal);
+
+        // Note: cos1**2 can be greater than 1.0 because of rounding errors -> min
+        double sin1 = std::sqrt(1.0 - std::min(1.0, cos1 * cos1));
+        sin1 = (scalarProd > 0) ? sin1 : -sin1;
+
+        auto* slip1 = local.layer->var(drDescr->slip1);
+        auto* slip2 = local.layer->var(drDescr->slip2);
+
+        slipVectors(DirectionID::Strike, level, i) =
+            cos1 * slip1[local.ltsId][local.nearestGpIndex] -
+            sin1 * slip2[local.ltsId][local.nearestGpIndex];
+
+        slipVectors(DirectionID::Dip, level, i) = sin1 * slip1[local.ltsId][local.nearestGpIndex] +
+                                                  cos1 * slip2[local.ltsId][local.nearestGpIndex];
+      }
+      this->outputSpecifics(outputData, level, i);
     }
 
-    auto& peakSlipsRate = std::get<VariableID::PeakSlipRate>(outputData.vars);
-    if (peakSlipsRate.isActive) {
-      peakSlipsRate(level, i) = peakSR[ltsId][nearestGaussPoint];
+    if (type == OutputType::AtPickpoint) {
+      outputData.cachedTime[outputData.currentCacheLevel] = time;
+      outputData.currentCacheLevel += 1;
     }
+  }
+}
+
+void Base::computeLocalStresses() {
+  real pFactorPlus = local.waveSpeedsPlus->pWaveVelocity * local.waveSpeedsPlus->density;
+  real pFactorMinus = local.waveSpeedsMinus->pWaveVelocity * local.waveSpeedsMinus->density;
+
+  real sFactorPlus = local.waveSpeedsPlus->sWaveVelocity * local.waveSpeedsPlus->density;
+  real sFactorMinus = local.waveSpeedsMinus->sWaveVelocity * local.waveSpeedsMinus->density;
+
+  real norDivisor = 1.0 / (pFactorMinus + pFactorPlus);
+  real shearDivisor = 1.0 / (sFactorMinus + sFactorPlus);
+  real uVelDivisor = 1.0 / (pFactorPlus);
+
+  auto diff = [this](int i) {
+    return this->local.faceAlignedValuesMinus[i] - this->local.faceAlignedValuesPlus[i];
+  };
+
+  local.xyStress = local.faceAlignedValuesPlus[3] +
+                   ((diff(3) + sFactorMinus * diff(7)) * sFactorPlus) * shearDivisor;
```

Comment:
Where is this formula is the current master branch? I would like to double-check it.

Reply:
Here is it: https://github.com/SeisSol/SeisSol/blob/acde11bb497ad9021636e804738e1380094e2c17/src/ResultWriter/faultoutput.f90#L423-L430


```
NorDivisor   = 1.0D0/(w_speed_neig(1)*rho_neig+w_speed(1)*rho)
ShearDivisor = 1.0D0/(w_speed_neig(2)*rho_neig+w_speed(2)*rho)
UVelDivisor  = 1.0D0/(w_speed(1)*rho)

LocXYStress = SideVal(4) + (((SideVal2(4)-SideVal(4)) + w_speed_neig(2)*rho_neig*(SideVal2(8)-SideVal(8)))* w_speed(2)*rho ) * ShearDivisor

LocXZStress = SideVal(6) + (((SideVal2(6)-SideVal(6)) + w_speed_neig(2)*rho_neig*(SideVal2(9)-SideVal(9)))* w_speed(2)*rho ) * ShearDivisor
```

Reply:
ok thx, looks good then!